language-plutus-core/test/Generators.hs

```diff
@@ -20,7 +22,48 @@ genTyName = TyName <$> genName
 genName :: MonadGen m => m (Name ())
 genName = Name () <$> name' <*> int'
     where int' = Unique <$> Gen.int (Range.linear 0 3000)
-          name' = BSL.fromStrict <$> Gen.utf8 (Range.linear 1 20) Gen.lower
+          name' = Gen.filter (not . isKw) (BSL.fromStrict <$> Gen.utf8 (Range.linear 1 20) Gen.lower)
+          isKw "abs"                      = True
```

Comment:
Our set of built-in names changes over time, so it's convenient to have them stored somewhere for cases when we want to do something over all of the built-in names. Which is why we have `allBuiltinNames` in `Language.PlutusCore.Lexer.Type`. So I'd define it as

```haskell
isKw s = elem s $
    [ "size", "type"                                              -- kind keywords
    , "ifix", "all", "bytestring", "integer"                      -- type keywords
    , "abs", "lam", "iwrap", "unwrap", "con", "builtin", "error"  -- term keywords
    , "program"                                                   -- other
    ] ++ map pretty allBuiltinNames
```

Reply:
But do we need to treat built-in names as keywords? @michaelpj

Reply:
I don't believe that we should, but the parser currently does so.

Reply:
We could also modify the lexer, yes. I chose the current setup to avoid expensive comparisons between strings, but it may be a marginal savings.

Reply:
Well, that doesn't need to block this, what @effectfully suggested seems like a nice improvement regardless.

Reply:
Yes, modifying the lexer is the correct approach.